src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] The `die` method uses a nested loop (indexOf inside filter) which can be inefficient. Consider using the `filter` method directly to remove dead animals from the `Animal.alive` array without the need for an `indexOf` check. This will help to avoid unnecessary complexity and potential performance issues.